test/no-default-parameter-options.js

```diff
@@ -0,0 +1,143 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/no-default-parameter-options';
+
+const ruleTester = avaRuleTester(test, {
+	parserOptions: {
+		ecmaVersion: 2018
+	}
+});
+
+const error = {
+	message: 'Use object spreading instead of passing default parameters with an object'
+};
+
+ruleTester.run('no-default-parameter-options', rule, {
+	valid: [
+		'let foo;',
+		'const foo = options => {};',
+		'const foo = (options = {}) => {};',
+		'const foo = (options = null) => {};',
+		'const foo = (options = undefined) => {};',
+		'const foo = ({a = true, b = true}) => {};',
+		'const foo = (bar = {a: false}) => {};',
+		'const foo = (bar = {a: false, b: true}) => {};',
+		'const foo = (bar = {}) => {};',
+		'function foo(options = {}) {}',
+		'function foo(options) {}',
+		`const foo = options => {
+			options = {
+				a: false,
+				b: true,
+				...options
+			};
+		};`,
+		`const fooDefaults = {};
+		const foo = (options = fooDefaults) => {};`,
+		`const fooDefaults = {
+			a: false,
+			b: true
+		};`
+	],
+	invalid: [
+		{
+			code: 'const foo = (options = {a: false}) => {};',
+			output: `const foo = (options) => {
+			options = {
+				a: false,
+				...options
+			};
+		};`,
+			errors: [error]
+		},
+		{
+			code: `const foo = (options = {a: true, b: false}) => {
+				bar();
+				baz = foo;
+			};`,
+			output: `const foo = (options) => {
+			options = {
+				a: true,
+b: false,
+				...options
+			};
+
+bar();
+baz = foo;
+		};`,
+			errors: [error]
+		},
+		{
+			code: 'const foo = (options = {a: false, b: 1, c: "test", d: null}) => {};',
+			output: `const foo = (options) => {
+			options = {
+				a: false,
+b: 1,
+c: "test",
+d: null,
```

Comment:
Please fix the indentation here

Reply:
This is necessary for the tests to work. Just like [this rule](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/test/no-unused-properties.js). But I'll see if I can come up with a solution that corrects the indentation.

Reply:
Oh, I see that your fixer generates the `options = {...}` statement without indentation, and this is probably fine as `indent` rule would probably fix the indentation (but I have not checked this). But why are you referring to [`no-unused-properties` tests](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/test/no-unused-properties.js) then? It has no fixer and no tests with `output` key.

Reply:
Yeah, I'm not sure whether it makes sense to make the fixer more complicated and error prone when it's being auto-indented automatically. I referred to the other rule because I thought the indentation also mattered in that rule's tests (which apparently doesn't 😏).

I could probably count the amount of spaces before the first statement and append those spaces to the next line. However I'm still waiting for @sindresorhus take on the question I asked [here](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/208#issuecomment-465053487).